File: C-Mandarin_Orange.go
package main
import (
	"fmt"
	"sort"
)
func min(list []int) int {
	sort.Sort(sort.IntSlice(list))
	return list[0]
}

func main() {
	var n int
	fmt.Scan(&n)
	A := make([] int, n)
	for i := 0; i < n; i ++ {
		fmt.Scan(&A[i])
	}
	ans := 0
	for i := 0; i < n; i ++ {
		x := A[i]
		for j := i; j < n; j ++ {
			if A[j] < x {
				x = A[j]
			}
			condidate := x * (j - i + 1)
			if ans < condidate {
				ans = condidate
			}
		}
	}
	fmt.Println(ans)

}

File: C-Mandarin_Orange.py
n = int(input())
A = list(map(int, input().split()))
ans = 0

for i in range(n):
    x = A[i]
    for j in range(i, n):
        if x > A[j]:
            x = A[j]
        condidate = x * (j - i + 1)
        
        if condidate > ans:
            ans = condidate
        

print(ans)


